src/amo/components/Collection/index.js

```diff
@@ -124,17 +134,49 @@ export class CollectionBase extends React.Component<Props> {
     return `/collections/${params.user}/${params.slug}/`;
   }
 
+  editCollection = (event: SyntheticEvent<any>) => {
+    const { dispatch } = this.props;
+    event.preventDefault();
+    event.stopPropagation();
+    dispatch(openFormOverlay(COLLECTION_OVERLAY));
+  }
+
+  editCollectionLink() {
+    const { _config, hasEditPermission, i18n } = this.props;
+
+    if (!hasEditPermission) {
```

Comment:
I feel like this check might be better in the `render()` method as it wasn't clear to me and it would prevent this method from even being called...

Reply:
Good call.